### PR TITLE
GRPC improvements for unix-type network

### PR DESCRIPTION
**Pull request by VladoLavor**
* the plugin creates a directory for unix domain socket file if does not exist
* removes old socket file if exists
* ability to set permissions

### CODE REVIEW

File: rpc/grpc/listen_and_serve.go [hunk 1]
```diff
@@ -41,9 +47,31 @@ func ListenAndServeGRPC(config *Config, grpcServer *grpc.Server) (netListener ne
 	if socketType == "" {
 		socketType = "tcp"
 	}
-	netListener, err = net.Listen(socketType, config.Endpoint)
-	if err != nil {
-		return nil, err
+
+	if socketType == "unix" || socketType == "unixpacket" {
+		permissions, err := getUnixSocketFilePermissions(config.Permission)
+		if err != nil {
```
Comment (edwarnicke) [inline rpc/grpc/listen_and_serve.go]: We should have default permission that we use if the user doesn't specify.

Comment (edwarnicke) [reply rpc/grpc/listen_and_serve.go]: Ah wait, you do that further down :)

File: rpc/grpc/listen_and_serve.go [hunk 2]
```diff
@@ -65,3 +93,42 @@ func ListenAndServeGRPC(config *Config, grpcServer *grpc.Server) (netListener ne
...
+		}
+		return os.FileMode(mode), nil
+	} else {
+		return os.ModePerm, nil
+	}
+}
+
+// Check old socket file/directory of the unix domain socket. Remove old socket file if exists or create the directory
+// path if does not exist.
+func checkUnixSocketFileAndDirectory(endpoint string) error {
```
[the remark this answers is not in the document]
Comment (VladoLavor) [reply rpc/grpc/listen_and_serve.go]: Well, contiv guys require it.  I'll make it configurable so everyone can be happy. By default, the existing socket file will NOT be removed, but you can force it in grpc.conf if you need.

File: rpc/grpc/config.go [hunk 3]
```diff
@@ -32,6 +32,9 @@ type Config struct {
 	// Three or four-digit permission setup for unix domain socket file (if used)
 	Permission int `json:"permission"`
 
+	// If set and unix type network is used, the existing socket file will be always removed and re-created
```
Comment (edwarnicke) [inline rpc/grpc/config.go]: +1

**Pull request merged**